src/main/java/com/google/api/codegen/transformer/java/JavaGapicSampleAppTransformer.java

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.java;
+
+import com.google.api.codegen.InterfaceView;
+import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.transformer.*;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.codegen.viewmodel.testing.SmokeTestClassView;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Responsible for producing sample application related views for Java GAPIC clients */
+public class JavaGapicSampleAppTransformer implements ModelToViewTransformer {
+
+  // Reuse the smoke test template since the sample application is almost the same as the smoke test class.
+  // (with a few exceptions such as class name)
+  private static String SMOKE_TEST_TEMPLATE_FILE = "java/smoke_test.snip";
```

Comment:
Maybe we should rename the snippet file to something more generic so people don't expect it to only be used for smoke tests.